charmcraft/utils.py

```diff
@@ -149,18 +160,21 @@ def get_os_platform(filepath=pathlib.Path("/etc/os-release")):
     machine = platform.machine()
 
     if system == "Linux":
-        os_release = {}
         try:
-            with filepath.open("r", encoding='utf-8') as f:
-                for line in f:
-                    if "=" in line:
-                        key, value = line.rstrip().split("=", 1)
-                        if value[0] == value[-1] == '"':
-                            value = value[1:-1]
-                        os_release[key] = value
+            with filepath.open("rt", encoding='utf-8') as fh:
+                lines = fh.readlines()
         except FileNotFoundError:
             logger.debug("Unable to locate 'os-release' file, using default values")
-        finally:
+        else:
+            os_release = {}
+            for line in lines:
+                line = line.strip()
+                if not line or line.startswith('#') or '=' not in line:
+                    continue
+                key, value = line.rstrip().split("=", 1)
+                if value[0] == value[-1] == '"':
```

Comment:
single quotes are also allowed by spec.

Reply:
Added a bunch of new tests for this case and a slight improvement here.